# PEDS example: Reaction--diffusion surrogate using a diffusion equation coarse solver

This example of PEDS shows how to use the code from: *Pestourie, Raphaël, et al. "Physics-enhanced deep surrogates for PDEs."*  In this example notebook, we illustrate PEDS for the surrogate of the reaction--diffusion equation's flux where the approximate solver is a coarse diffusion equation's solver. For this example, the input is the side length of 16 holes in a material.

NB: since the approximate solver is the same for the reaction-diffusion and the diffusion surrogates, training the surrogate models with the 25 holes structures for the reaction-diffusion equation necessitates only to change the dataset from `X/y_fisher16_small.csv` to `X/y_fourier16_small.csv`.
    
## Note on implementation

In this notebook, we run the training on a single processor and J=1.

However, the implementation of this code is meant to be parallel on multiple processors, where the element of a batch is computed in parallel over J groups of CPUs (J is the number models in the ensemble). For example, with a julia file `train_PEDS.jl`containing functions calls similart to the code in the cells below, the command to run large-scale training on a cluster would be:
    
`mpiexec -n 320 julia train_PEDS.jl` 

In [1]:
#initialize functions

include("../src/PEDS.jl")
include("../src/fourier_solver.jl")

##training functionalities
function train_distributed!(comm, commModel, commLeader, mloglik, m, loss, ps, loader, opt, validation_fes; logging=false)

    train_lossmean = 0.
    k = 0
    for d in loader
        X,y = d #change for Fourier
        train_loss, back = Zygote.pullback(() -> loss(commModel, mloglik, m, X, y), ps)
        train_lossmean *= k
        k+=1
        train_lossmean += train_loss
        train_lossmean /= k

        gs = back(Float32(1.))
        if debug && isleader
            if isnan(train_loss)
                @show (model_color, train_loss)
            end
        end
        for x in ps
            gs[x][:] .= (sum_reduce(commModel, (gs[x][:]) ))
            if debug && isleader
                if any(isnan.(gs[x][:]))
                    @show (model_color, train_loss)
                    @show (length(x), length(findall(isnan,x)), length(findall(isnan,gs[x][:])))
                end

                if any(isinf.(gs[x][:]))
                    @show (model_color, train_loss)
                    @show (length(x), length(findall(isinf,x)), length(findall(isinf,gs[x][:])))
                end
            end
        end

        if debug && isleader
            for p_ in ps
                if any(isnan.(p_))
                    @show (model_color, "before update")
                end
            end
        end
        Flux.Optimise.update!(opt, ps, gs)
        if debug && isleader
            for p_ in ps
                if any(isnan.(p_))
                    @show (model_color, "after update")
                end
            end
        end
    end

    # logging && push!(validation_fes, dFE(comm, commModel, commLeader, m))
    #changing for fourier
    if logging
        if MPI.Comm_rank(comm) == 0
            @show train_lossmean
            push!(validation_fes, train_lossmean)
        end
        MPI.Barrier(comm)
        valid_lossMSE = lossvalid(comm, commModel, commLeader, mloglik, m)
        valid_lossFE = dFE_1d(comm, commModel, commLeader, m)
        if MPI.Comm_rank(comm) == 0
            push!(validation_fes, valid_lossMSE)
            push!(validation_fes, valid_lossFE)
        end
    end
end

function pore(p, resolution; lowval=0.1)

    if resolution % 2==0
        inner = p/2*resolution ÷ 1
        w =  (p*resolution)/2 % 1

        isqb = Int((resolution/2+inner) ÷ 1) # square index to the right
        isqa = Int((resolution/2-inner) ÷ 1) # square index to the left
    else
        inner = p/2*resolution ÷ 1
        w = ((p*resolution)/2 % 1)
        w*= resolution ==1 ? 2 : 1
        isqb = Int((resolution/2+inner) ÷ 1) # square index to the right
        isqa = Int((resolution/2-inner) ÷ 1)+1 # square index to the left
    end

    begin
        A1 = ones((resolution, resolution))
        A2 = ones((resolution, resolution))
        A1[isqa+1:isqb, isqa+1:isqb] .= lowval

        if isqa==0 && w==0
            return A1
        elseif isqa==0 && w!=0
            return A1 .* (1-w)
        else
            A2[isqa:isqb+1, isqa:isqb+1] .= lowval
            return A1 .* (1-w) .+ A2 .* w
        end
    end

end

function generatepores(ps, resolution)
    n = sqrt(length(ps))
    N = resolution÷n
    @assert n == Int(n)
    @assert N == Int(N)
    n = Int(n)
    N = Int(N)
    pores = zeros((resolution, resolution))
    for (k,p) in enumerate(ps)
        i = (k-1)÷n
        j = (k-1)%n
        pores[i*N+1:(i+1)*N, j*N+1:(j+1)*N] = pore(p, N)
    end
    return pores
end

ChainRules.@non_differentiable generatepores(ps, resolution)

In [2]:
startp, endp = 7, 7
PEDSname = "PEDS"
equation = "exampleFisher"

ninit = 64
Jval = 1

cwval = 0.05
learningrate = 5e-5
nnodes =128

optimizerfunc, optname = Flux.ADAM, "ADAM"

number_epoch = 200
batchsize = 64

datalimitvalid = 2^10

Ks = Int[2^i for i=startp:endp]

coarseresolution = 4
Lx = Ly = 1
nin=Int(coarseresolution^2)
coarse_sim = setsimulation(Lx, Ly, coarseresolution);
nout = coarseresolution^2
const avgOpcoarse = create_cOp(coarseresolution);

endname = "$(equation)_$(2^startp)to$(2^endp)_ne$(number_epoch)_lr$(learningrate)_bs$(batchsize)_opt$(optname)_hn$(nnodes)_coarse$(coarseresolution)_seed";

##loading data
X = readdlm("../data/X_fisher$(coarseresolution^2)_small.csv", ',')
y = readdlm("../data/y_fisher$(coarseresolution^2)_small.csv", ',')[:];

Xv = X[:, 1:1024] #valid set
Xtest = X[:, end-1023:end] #test set
Xt = X[:, 1025:end]

yv = y[1:1024] #valid set
ytest = y[end-1023:end] #test set
yt = y[1025:end];

In [3]:
##Definition of problem constants
const debug = false
const drv = DataRunner(Xv, yv, [1]);
const al = ALstruct(J=Jval, Nvalid=datalimitvalid, batchsize=batchsize);
const valid = initvalid(al, drv) #validation loader

const drtest = DataRunner(Xtest, ytest, [1]);
const test = initvalid(al, drtest) #validation loader

##setup MPI and random
const comm = MPI.COMM_WORLD
const model_color = MPI.Comm_rank(comm)%al.J
const commModel = MPI.Comm_split(comm, model_color, 0)
const isleader = MPI.Comm_rank(commModel) == 0
const commLeader = MPI.Comm_split(comm, isleader, 0)
debug && print("Comm rank=$(MPI.Comm_rank(comm)), commModel rank = $(MPI.Comm_rank(commModel)), commLeader rank = $(MPI.Comm_rank(commLeader))\n")

Random.seed!(2138*(model_color+1)) #alter seed for different groups

function lossvalid(comm, commModel, commLeader, mloglik, m)
    valid_lossmean = 0.
    k = 0
    testmode!(m, true)
    for d in valid
        X,y=d
        valid_loss = dMSE(commModel, mloglik, m, X, y)

        valid_lossmean *= k
        k+=1
        valid_lossmean += valid_loss
        valid_lossmean /= k
    end
    testmode!(m, false)
    # MPI.Comm_rank(commModel)== 0 && @show valid_lossmean
    valid_lossmean = sum_reduce(commLeader, valid_lossmean) / al.J
    MPI.Comm_rank(comm) == 0 && @show valid_lossmean
    return valid_lossmean
end

function dFE_1d(comm, commModel, commLeader, m; valid=valid) """dFE computes the FE using parallelization over the batch with MPI"""
    evalsr = zeros(al.Nvalid)
    FE = 0.
    j=0
    ys = []
    testmode!(m, true)
    for (x, y) in valid
                rp = m(x[:,1+MPI.Comm_rank(commModel):MPI.Comm_size(commModel):length(y)])[:]
        evalsr[j*length(y)+1+MPI.Comm_rank(commModel):MPI.Comm_size(commModel):j*length(y)+length(y)] .= rp
        push!(ys, y...)
        j+=1
    end
    testmode!(m, false)
    evalsrModel = sum_reduce(commModel, evalsr)
    evalsr = sum_reduce(commLeader, evalsrModel) / al.J
    if MPI.Comm_rank(comm) == 0
        FE = norm(evalsr - ys)/norm(ys)
        MPI.Comm_rank(comm) == 0 && @show FE
    end
    return FE
end

dFE_1d (generic function with 1 method)

# PEDS

In [4]:
namemodel = "local_MSE_parallel_PEDSparameterized_$endname"

kval =  Ks[1]
##define same AL parameters for all workers
al1 = ALstruct(Ninit=ninit+8*kval, T=0, η=1e-3, J=Jval, ne=number_epoch, batchsize=batchsize);
if MPI.Comm_rank(comm) == 0
    @show kval
    @show 256+8*kval
end
##ititialize DataRunner and DataSet
dr = DataRunner(Xt, yt, [1]);
ds = DataSet()
validation_fes = []
mgen =  Chain(Dense(nin, nnodes, relu),
Dropout(0.5),
Dense(nnodes, nnodes,relu),
Dropout(0.5),
Dense(nnodes, coarseresolution^2, hardtanh),
x-> @. x*0.9/2 + 0.45 + .1)

cw=[cwval]

m(p) = begin
    generatedgeom = mgen(p)
    w = @. max(0, min(1, cw)) # harder function.
    @views map(i->Float32(targetfunc(avgOpcoarse * (w .* generatedgeom[:,i] + (1 .-w) .* generatepores(p[:, i],coarseresolution)[:]), sim=coarse_sim)), 1:length(p)÷ nin)
end
mloglik(p) = 1
ps = Flux.params(mgen, cw)


loader = initloader(al1, dr, ds);
opt = ADAM(al1.η)

if MPI.Comm_rank(comm)==0
    @time Flux.@epochs al1.ne train_distributed!(comm, commModel, commLeader, mloglik, m, dHuber, ps, loader, opt, validation_fes, logging=true)
else
    Flux.@epochs al1.ne train_distributed!(comm, commModel, commLeader, mloglik, m, dHuber, ps, loader, opt, validation_fes, logging=true)
end


# ##save models and validation FEs
# if isleader
#     BSON.@save "$(namemodel)_K$(kval)_mgen$(model_color).bson" mgen
#     BSON.@save "$(namemodel)_K$(kval)_cw$(model_color).bson" cw
# end

# if MPI.Comm_rank(comm) == 0
#     writedlm("$(namemodel)_K$(kval)_trainingMSE.csv", validation_fes, ',')
# end

# # evaluate and show the error on the test set
# valid_lossmean = dFE_1d(comm, commModel, commLeader, m)
# if MPI.Comm_rank(comm) == 0
#     @show valid_lossmean
#     writedlm("$(namemodel)_K$(kval)_endvalidFE.csv", [valid_lossmean], ',')
# end

kval = 128
256 + 8kval = 1280


┌ Info: Epoch 1
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 7.042909392808912e-5
valid_lossmean = 0.0061226879368676875
FE = 0.16059397858016888


┌ Info: Epoch 2
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.341972747794767e-5
valid_lossmean = 0.005388577354486344
FE = 0.15065906088131042


┌ Info: Epoch 3
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.083447258975376e-5
valid_lossmean = 0.005177806667255543
FE = 0.14768320553127198


┌ Info: Epoch 4
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.980984259333528e-5
valid_lossmean = 0.005090293616193124
FE = 0.14642984817219407


┌ Info: Epoch 5
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.921202250587817e-5
valid_lossmean = 0.004968439193629799
FE = 0.14466657002740377


┌ Info: Epoch 6
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.835522666372692e-5
valid_lossmean = 0.004673165476691721
FE = 0.14030197190975863


┌ Info: Epoch 7
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.704063897728455e-5
valid_lossmean = 0.004347676347153957
FE = 0.13532773044922183


┌ Info: Epoch 8
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.4947929677053264e-5
valid_lossmean = 0.003886060470171431
FE = 0.12794195195845431


┌ Info: Epoch 9
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.163956278880603e-5
valid_lossmean = 0.0031476350730569474
FE = 0.115146404927782


┌ Info: Epoch 10
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.7095611623320426e-5
valid_lossmean = 0.002358693516703302
FE = 0.09967676895760252


┌ Info: Epoch 11
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.083827752877201e-5
valid_lossmean = 0.001966145911921746
FE = 0.09100517045735859


┌ Info: Epoch 12
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.919313746824996e-5
valid_lossmean = 0.0019195080097244416
FE = 0.08991935013145805


┌ Info: Epoch 13
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.752761997148753e-5
valid_lossmean = 0.0017169324158489143
FE = 0.08504226043179887


┌ Info: Epoch 14
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.631052041743131e-5
valid_lossmean = 0.0016501605306909848
FE = 0.08337220596860859


┌ Info: Epoch 15
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.5530394120715786e-5
valid_lossmean = 0.001553505006217098
FE = 0.08089366714737797


┌ Info: Epoch 16
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.4435393422584764e-5
valid_lossmean = 0.0014939579332626008
FE = 0.07932815973055747


┌ Info: Epoch 17
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.3257536058641144e-5
valid_lossmean = 0.0014433852139295476
FE = 0.07797391156616122


┌ Info: Epoch 18
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.472672791861738e-5
valid_lossmean = 0.0014199350169320237
FE = 0.07733790972459474


┌ Info: Epoch 19
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.391546200377919e-5
valid_lossmean = 0.0013644007883674262
FE = 0.07581046770506616


┌ Info: Epoch 20
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.2146421885895326e-5
valid_lossmean = 0.0013039150270218374
FE = 0.07411102849559234


┌ Info: Epoch 21
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.1937128883573094e-5
valid_lossmean = 0.0012880046259930384
FE = 0.07365748848911227


┌ Info: Epoch 22
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.25370652559855e-5
valid_lossmean = 0.0013208075664063353
FE = 0.07458954693896408


┌ Info: Epoch 23
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.112817661204202e-5
valid_lossmean = 0.001216905516734457
FE = 0.0715956481571281


┌ Info: Epoch 24
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.109170353298982e-5
valid_lossmean = 0.00119540360303661
FE = 0.07096030530904457


┌ Info: Epoch 25
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.142691411382063e-5
valid_lossmean = 0.0012015367406809617
FE = 0.07114210688837906


┌ Info: Epoch 26
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.0281680038620237e-5
valid_lossmean = 0.0011342873406746774
FE = 0.06912254783276449


┌ Info: Epoch 27
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.984921819902868e-5
valid_lossmean = 0.0010779142933149278
FE = 0.06738299502988808


┌ Info: Epoch 28
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.993270090853858e-5
valid_lossmean = 0.0010583791527727408
FE = 0.06676960908139692


┌ Info: Epoch 29
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.9894911468066525e-5
valid_lossmean = 0.0010718991444063468
FE = 0.06719472134670865


┌ Info: Epoch 30
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.864828279094486e-5
valid_lossmean = 0.0010011445861503553
FE = 0.06493914954679975


┌ Info: Epoch 31
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.967875955936962e-5
valid_lossmean = 0.0009795905895612642
FE = 0.0642362969907142


┌ Info: Epoch 32
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.921738259631252e-5
valid_lossmean = 0.0010310560949010077
FE = 0.06590211339357774


┌ Info: Epoch 33
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.866007397669586e-5
valid_lossmean = 0.0010005661767865993
FE = 0.06492038760196274


┌ Info: Epoch 34
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.8206242465340144e-5
valid_lossmean = 0.0009422055266968312
FE = 0.06299861747446253


┌ Info: Epoch 35
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.8371132238923787e-5
valid_lossmean = 0.0010557798430233176
FE = 0.06668756778914943


┌ Info: Epoch 36
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.7370152562399583e-5
valid_lossmean = 0.0009156917267437115
FE = 0.06210589723117755


┌ Info: Epoch 37
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.7519171639851294e-5
valid_lossmean = 0.0008852727890906387
FE = 0.06106561746627038


┌ Info: Epoch 38
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.729959010063028e-5
valid_lossmean = 0.0008548335712943079
FE = 0.060006594443509885


┌ Info: Epoch 39
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.706825180542091e-5
valid_lossmean = 0.0008500098046640113
FE = 0.059837048393017574


┌ Info: Epoch 40
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.666440892325677e-5
valid_lossmean = 0.0008852487439897283
FE = 0.0610647881518925


┌ Info: Epoch 41
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.689407229374232e-5
valid_lossmean = 0.0008505645635591223
FE = 0.059856571532772365


┌ Info: Epoch 42
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6391381072430117e-5
valid_lossmean = 0.0008321773884664083
FE = 0.05920605860783074


┌ Info: Epoch 43
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5537738267365586e-5
valid_lossmean = 0.0007912562400174896
FE = 0.05773202209148398


┌ Info: Epoch 44
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6073494630490706e-5
valid_lossmean = 0.0007672071148012767
FE = 0.05684791044485583


┌ Info: Epoch 45
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6810814326965796e-5
valid_lossmean = 0.0007983019636666077
FE = 0.05798848918588934


┌ Info: Epoch 46
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5912091410253345e-5
valid_lossmean = 0.0008282910294165985
FE = 0.059067647448534356


┌ Info: Epoch 47
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5228021257524762e-5
valid_lossmean = 0.0007643127525268543
FE = 0.05674057702610216


┌ Info: Epoch 48
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6020633635529857e-5
valid_lossmean = 0.0007785812862357821
FE = 0.057267757280481335


┌ Info: Epoch 49
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5288587085345364e-5
valid_lossmean = 0.0007903063799216542
FE = 0.057697359608927544


┌ Info: Epoch 50
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5210710025656675e-5
valid_lossmean = 0.0007808607250173991
FE = 0.05735152691705328


┌ Info: Epoch 51
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.492614295098781e-5
valid_lossmean = 0.0007892146532595821
FE = 0.057657494364313146


┌ Info: Epoch 52
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4013810113925943e-5
valid_lossmean = 0.0007766148550401109
FE = 0.05719539213252625


┌ Info: Epoch 53
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4582686915864443e-5
valid_lossmean = 0.0007088666042129161
FE = 0.05464374467681999


┌ Info: Epoch 54
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.5184556590177317e-5
valid_lossmean = 0.0007041722098080374
FE = 0.05446250790192662


┌ Info: Epoch 55
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.441542931446774e-5
valid_lossmean = 0.0007500445157628186
FE = 0.056208463681073406


┌ Info: Epoch 56
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4876408130327078e-5
valid_lossmean = 0.0006845860351812106
FE = 0.053699743716470974


┌ Info: Epoch 57
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3992118385103214e-5
valid_lossmean = 0.0007287533997281449
FE = 0.05540494040209532


┌ Info: Epoch 58
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4372729169339146e-5
valid_lossmean = 0.0006533650682832668
FE = 0.05246095000724099


┌ Info: Epoch 59
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.359006369015686e-5
valid_lossmean = 0.0006421783765893595
FE = 0.0520099018390112


┌ Info: Epoch 60
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.309988495128123e-5
valid_lossmean = 0.0007018533844715727
FE = 0.05437276197058926


┌ Info: Epoch 61
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.367494955776498e-5
valid_lossmean = 0.0006633007290012228
FE = 0.052858329329208975


┌ Info: Epoch 62
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3539590507162793e-5
valid_lossmean = 0.0006569378313634661
FE = 0.052604189233949335


┌ Info: Epoch 63
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.385644380304605e-5
valid_lossmean = 0.0006927053741816095
FE = 0.054017250376262385


┌ Info: Epoch 64
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2423069192626216e-5
valid_lossmean = 0.0006487476359408131
FE = 0.05227524648867901


┌ Info: Epoch 65
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.292853621976707e-5
valid_lossmean = 0.0006324384670913604
FE = 0.05161397817298494


┌ Info: Epoch 66
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.254029322024461e-5
valid_lossmean = 0.0005973645089406693
FE = 0.05016235338130769


┌ Info: Epoch 67
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.325498983320763e-5
valid_lossmean = 0.000618497151735143
FE = 0.05104192524593458


┌ Info: Epoch 68
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2035825507208877e-5
valid_lossmean = 0.0006462981624328135
FE = 0.05217646545313053


┌ Info: Epoch 69
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.276060735086799e-5
valid_lossmean = 0.0005827941734021072
FE = 0.04954682111622422


┌ Info: Epoch 70
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2479643295897115e-5
valid_lossmean = 0.0005689070902392667
FE = 0.048952949997258466


┌ Info: Epoch 71
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2408973929900184e-5
valid_lossmean = 0.000559177039193836
FE = 0.048532522018271136


┌ Info: Epoch 72
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2030465819053893e-5
valid_lossmean = 0.0005997092493018494
FE = 0.05026070413888951


┌ Info: Epoch 73
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2289754918629253e-5
valid_lossmean = 0.0005510778025509795
FE = 0.048179762422113796


┌ Info: Epoch 74
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1728366631076494e-5
valid_lossmean = 0.0005773856865703505
FE = 0.049316381314600845


┌ Info: Epoch 75
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.209349910908387e-5
valid_lossmean = 0.0005301043179558019
FE = 0.047254031453319245


┌ Info: Epoch 76
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1779324974783556e-5
valid_lossmean = 0.0005382586569187024
FE = 0.04761608741743425


┌ Info: Epoch 77
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.156018284089361e-5
valid_lossmean = 0.0005539878112393981
FE = 0.04830680339964188


┌ Info: Epoch 78
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2095675533464116e-5
valid_lossmean = 0.0005179836733440894
FE = 0.04671068441653301


┌ Info: Epoch 79
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0946643357304418e-5
valid_lossmean = 0.0005934482165657575
FE = 0.04999765203166413


┌ Info: Epoch 80
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.14098645413275e-5
valid_lossmean = 0.0005156132984933478
FE = 0.046603684140170056


┌ Info: Epoch 81
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1620082771046127e-5
valid_lossmean = 0.0006231807376176555
FE = 0.05123481891951309


┌ Info: Epoch 82
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.036405162621767e-5
valid_lossmean = 0.000655083014953868
FE = 0.052529874681349814


┌ Info: Epoch 83
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1746798741222138e-5
valid_lossmean = 0.0005770046887976698
FE = 0.04930010750271976


┌ Info: Epoch 84
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0773554130952996e-5
valid_lossmean = 0.0005791084990168713
FE = 0.04938990200753646


┌ Info: Epoch 85
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0395511434996147e-5
valid_lossmean = 0.0004917396600391855
FE = 0.045511988756335865


┌ Info: Epoch 86
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1488547172202555e-5
valid_lossmean = 0.0005831677966404646
FE = 0.04956270054554409


┌ Info: Epoch 87
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0555244590556893e-5
valid_lossmean = 0.0004822053377181404
FE = 0.04506861394773445


┌ Info: Epoch 88
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0504698069651944e-5
valid_lossmean = 0.0004923234648927507
FE = 0.04553899719250691


┌ Info: Epoch 89
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.134578218179629e-5
valid_lossmean = 0.0005070247967865179
FE = 0.046213918591533114


┌ Info: Epoch 90
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0635648060145567e-5
valid_lossmean = 0.0005070083673508763
FE = 0.04621316983648512


┌ Info: Epoch 91
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9707708568187153e-5
valid_lossmean = 0.0005222991732185361
FE = 0.04690486219188493


┌ Info: Epoch 92
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0549788699435706e-5
valid_lossmean = 0.0005013046516579181
FE = 0.04595249139451132


┌ Info: Epoch 93
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.079480411538594e-5
valid_lossmean = 0.00048814350256549087
FE = 0.04534526576957173


┌ Info: Epoch 94
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9697044313735764e-5
valid_lossmean = 0.0005412619268925087
FE = 0.047748742082602345


┌ Info: Epoch 95
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0508757200866465e-5
valid_lossmean = 0.0005655840795359481
FE = 0.0488097724747642


┌ Info: Epoch 96
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.040489303744247e-5
valid_lossmean = 0.00048488332328696493
FE = 0.04519358766858761


┌ Info: Epoch 97
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.04491449742212e-5
valid_lossmean = 0.0005029180551875224
FE = 0.04602637895392445


┌ Info: Epoch 98
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0395302345782932e-5
valid_lossmean = 0.0005282745789343849
FE = 0.047172408571583474


┌ Info: Epoch 99
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0250666753138436e-5
valid_lossmean = 0.0005463033664849784
FE = 0.047970598125085036


┌ Info: Epoch 100
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0470587063573832e-5
valid_lossmean = 0.0005742697155530565
FE = 0.04918312870293651


┌ Info: Epoch 101
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.995257458243918e-5
valid_lossmean = 0.000526746838397876
FE = 0.04710414920163583


┌ Info: Epoch 102
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0519233057753937e-5
valid_lossmean = 0.0005318955558541032
FE = 0.047333800497547415


┌ Info: Epoch 103
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.94460053962435e-5
valid_lossmean = 0.00047575058584245744
FE = 0.044765955743584225


┌ Info: Epoch 104
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.963971801134175e-5
valid_lossmean = 0.0004735311746634709
FE = 0.04466141545489922


┌ Info: Epoch 105
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9464223077079062e-5
valid_lossmean = 0.0004513958611204153
FE = 0.043605069345430916


┌ Info: Epoch 106
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9799168465548292e-5
valid_lossmean = 0.00048566760055689177
FE = 0.045230122213121754


┌ Info: Epoch 107
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.980045191726819e-5
valid_lossmean = 0.00043666885624005035
FE = 0.042887852880923664


┌ Info: Epoch 108
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8755800440445797e-5
valid_lossmean = 0.00047408516903088907
FE = 0.044687532996420934


┌ Info: Epoch 109
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.92656065988576e-5
valid_lossmean = 0.00046060194375924295
FE = 0.04404748114521353


┌ Info: Epoch 110
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9433004413032753e-5
valid_lossmean = 0.0004884408009895974
FE = 0.04535907218508555


┌ Info: Epoch 111
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9603491856700973e-5
valid_lossmean = 0.000455830760378939
FE = 0.04381875253657252


┌ Info: Epoch 112
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.86892324261924e-5
valid_lossmean = 0.0004162405516605605
FE = 0.04187264444361334


┌ Info: Epoch 113
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8973465034402777e-5
valid_lossmean = 0.000409436812481743
FE = 0.04152901580557641


┌ Info: Epoch 114
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8903713699579513e-5
valid_lossmean = 0.000466265612006639
FE = 0.044317462760001106


┌ Info: Epoch 115
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8868943230668087e-5
valid_lossmean = 0.00042047087002149305
FE = 0.04208488569058643


┌ Info: Epoch 116
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.9352890405421934e-5
valid_lossmean = 0.00045738489290868137
FE = 0.04389338791373484


┌ Info: Epoch 117
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7898222871815293e-5
valid_lossmean = 0.0005445337738005143
FE = 0.0478928416264133


┌ Info: Epoch 118
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8774564924809115e-5
valid_lossmean = 0.0005077300895984621
FE = 0.04624605017331356


┌ Info: Epoch 119
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8290924580261872e-5
valid_lossmean = 0.0005418072304902782
FE = 0.04777278867156233


┌ Info: Epoch 120
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7898058204463873e-5
valid_lossmean = 0.0004748940218354295
FE = 0.044725638208907986


┌ Info: Epoch 121
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.883382197980994e-5
valid_lossmean = 0.00048056102847641123
FE = 0.04499170685428093


┌ Info: Epoch 122
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8335160102587136e-5
valid_lossmean = 0.0005142771560690406
FE = 0.046543261384803435


┌ Info: Epoch 123
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8664880369576772e-5
valid_lossmean = 0.0004809035810687373
FE = 0.04500773944888261


┌ Info: Epoch 124
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8300402470503374e-5
valid_lossmean = 0.0004637614287754916
FE = 0.044198294141150944


┌ Info: Epoch 125
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8487688765198712e-5
valid_lossmean = 0.00039489685822373415
FE = 0.04078495940863689


┌ Info: Epoch 126
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8506321562500886e-5
valid_lossmean = 0.00043222355636028784
FE = 0.042668994753219704


┌ Info: Epoch 127
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8201194842098016e-5
valid_lossmean = 0.0005022239099199486
FE = 0.045994604368878596


┌ Info: Epoch 128
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.858574281982014e-5
valid_lossmean = 0.0005429335614608919
FE = 0.04782241890246796


┌ Info: Epoch 129
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.773919389450238e-5
valid_lossmean = 0.000536075328293235
FE = 0.047519417160433106


┌ Info: Epoch 130
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8778590977359865e-5
valid_lossmean = 0.000541328060170447
FE = 0.047751659047774994


┌ Info: Epoch 131
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.8165969514720113e-5
valid_lossmean = 0.0004983774504159489
FE = 0.04581813285166916


┌ Info: Epoch 132
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7740143528441237e-5
valid_lossmean = 0.0005260475189629901
FE = 0.04707287062045539


┌ Info: Epoch 133
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.801752899938441e-5
valid_lossmean = 0.0004043013105593917
FE = 0.04126774798398025


┌ Info: Epoch 134
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.827414927072365e-5
valid_lossmean = 0.00043127357986591454
FE = 0.0426220782461193


┌ Info: Epoch 135
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.832377608050398e-5
valid_lossmean = 0.0005927906408010727
FE = 0.049969944174532


┌ Info: Epoch 136
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.784793287967921e-5
valid_lossmean = 0.0005416535983783801
FE = 0.0477660150864579


┌ Info: Epoch 137
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.791429393447136e-5
valid_lossmean = 0.0004968596589393562
FE = 0.04574831087290163


┌ Info: Epoch 138
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.727424636794758e-5
valid_lossmean = 0.00045228354697361646
FE = 0.04364792373914671


┌ Info: Epoch 139
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.779646047804548e-5
valid_lossmean = 0.0005085121238140262
FE = 0.04628165184318222


┌ Info: Epoch 140
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7864765888687706e-5
valid_lossmean = 0.0005309782144209552
FE = 0.04729296541832522


┌ Info: Epoch 141
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.744914004668224e-5
valid_lossmean = 0.00045515914072661977
FE = 0.043786459428863486


┌ Info: Epoch 142
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.766930127714697e-5
valid_lossmean = 0.00044774067123793385
FE = 0.043428163894219086


┌ Info: Epoch 143
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7850396197697155e-5
valid_lossmean = 0.0004939945063531433
FE = 0.04561621582485019


┌ Info: Epoch 144
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7582490059767704e-5
valid_lossmean = 0.0005022460703517413
FE = 0.04599561910456666


┌ Info: Epoch 145
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7621699885964213e-5
valid_lossmean = 0.00044999188797735257
FE = 0.04353720427509117


┌ Info: Epoch 146
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7782243035603043e-5
valid_lossmean = 0.00038496427639789785
FE = 0.040268774246081114


┌ Info: Epoch 147
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.693393966459508e-5
valid_lossmean = 0.00046144074545998397
FE = 0.04408757031202607


┌ Info: Epoch 148
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.708734044786244e-5
valid_lossmean = 0.0005090593423703003
FE = 0.04630654738421106


┌ Info: Epoch 149
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.753447926910157e-5
valid_lossmean = 0.0004325950999980423
FE = 0.04268733015839004


┌ Info: Epoch 150
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.773521684995718e-5
valid_lossmean = 0.00048066375732925604
FE = 0.04499651550414895


┌ Info: Epoch 151
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7386076829443977e-5
valid_lossmean = 0.00046274739639864005
FE = 0.044149947055076286


┌ Info: Epoch 152
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7570844152788127e-5
valid_lossmean = 0.0004391718151143909
FE = 0.04301059250822342


┌ Info: Epoch 153
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.699426273473466e-5
valid_lossmean = 0.00042556731118975437
FE = 0.04233916861708429


┌ Info: Epoch 154
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7416746394946945e-5
valid_lossmean = 0.0004419138591974757
FE = 0.04314465555514727


┌ Info: Epoch 155
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7801087783364532e-5
valid_lossmean = 0.0005250085957475923
FE = 0.0470263641104678


┌ Info: Epoch 156
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.796443430294211e-5
valid_lossmean = 0.0005323130331092037
FE = 0.04735237266732671


┌ Info: Epoch 157
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.747704698881685e-5
valid_lossmean = 0.0004837842447088027
FE = 0.04514233875884289


┌ Info: Epoch 158
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.736887218431941e-5
valid_lossmean = 0.00046202543569901915
FE = 0.04411549308626654


┌ Info: Epoch 159
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6508604764227383e-5
valid_lossmean = 0.00045134389773444065
FE = 0.04360255942846247


┌ Info: Epoch 160
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6751253192769542e-5
valid_lossmean = 0.00041361265455735444
FE = 0.041740255578349046


┌ Info: Epoch 161
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.734659922904497e-5
valid_lossmean = 0.0004926159180965415
FE = 0.04555252087088908


┌ Info: Epoch 162
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7672747635430778e-5
valid_lossmean = 0.0004029102602955885
FE = 0.041196693336004855


┌ Info: Epoch 163
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6860333862605193e-5
valid_lossmean = 0.00045315313764184527
FE = 0.04368986380178897


┌ Info: Epoch 164
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.645181193888985e-5
valid_lossmean = 0.0004161961994503621
FE = 0.041870413529548366


┌ Info: Epoch 165
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.697474521676299e-5
valid_lossmean = 0.000448677847124953
FE = 0.043473590360324374


┌ Info: Epoch 166
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7350706901094913e-5
valid_lossmean = 0.00044075826398510355
FE = 0.04308820745639779


┌ Info: Epoch 167
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.667533634646778e-5
valid_lossmean = 0.0004884943280706755
FE = 0.045361557514131755


┌ Info: Epoch 168
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6933444381193124e-5
valid_lossmean = 0.000442364852183547
FE = 0.043166665472627104


┌ Info: Epoch 169
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6999374257693624e-5
valid_lossmean = 0.0004961040412232201
FE = 0.04571351091853802


┌ Info: Epoch 170
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7217072718550855e-5
valid_lossmean = 0.0004305490067207143
FE = 0.04258625899237653


┌ Info: Epoch 171
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6456137220064246e-5
valid_lossmean = 0.0004328373149881673
FE = 0.04269927905050063


┌ Info: Epoch 172
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.7389361229978007e-5
valid_lossmean = 0.0004859238448205124
FE = 0.045242052627049084


┌ Info: Epoch 173
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6318490916818483e-5
valid_lossmean = 0.0004730477048340262
FE = 0.044638610239220645


┌ Info: Epoch 174
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6101195751066703e-5
valid_lossmean = 0.0005188674895607965
FE = 0.04675051778128933


┌ Info: Epoch 175
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.676613752277046e-5
valid_lossmean = 0.0005950913498317468
FE = 0.050066820676554394


┌ Info: Epoch 176
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.686651485943884e-5
valid_lossmean = 0.0005846092050506894
FE = 0.0496239144910761


┌ Info: Epoch 177
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.710972513161452e-5
valid_lossmean = 0.0005778426787720068
FE = 0.04933589404629609


┌ Info: Epoch 178
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6932093202432253e-5
valid_lossmean = 0.0004800041400020587
FE = 0.04496563043064469


┌ Info: Epoch 179
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.594411224784294e-5
valid_lossmean = 0.0005242380089890282
FE = 0.046991839722978615


┌ Info: Epoch 180
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.646070998559243e-5
valid_lossmean = 0.0005062224524983295
FE = 0.04617733837432658


┌ Info: Epoch 181
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6944383324643968e-5
valid_lossmean = 0.0004924749941228016
FE = 0.04554600473814585


┌ Info: Epoch 182
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.676707829222163e-5
valid_lossmean = 0.0004891540129128327
FE = 0.045392176330167774


┌ Info: Epoch 183
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6361145709667908e-5
valid_lossmean = 0.00047880086517261577
FE = 0.04490923512319557


┌ Info: Epoch 184
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6368830650596757e-5
valid_lossmean = 0.0004642044706925817
FE = 0.04421940092286598


┌ Info: Epoch 185
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6338807246316306e-5
valid_lossmean = 0.00046852750470065583
FE = 0.04442482653173586


┌ Info: Epoch 186
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6516065896129505e-5
valid_lossmean = 0.0005145277066585558
FE = 0.0465545977055522


┌ Info: Epoch 187
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.653964802808668e-5
valid_lossmean = 0.0004596431269791759
FE = 0.04400161131503617


┌ Info: Epoch 188
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6059530506080885e-5
valid_lossmean = 0.0004003979119186076
FE = 0.041068051420911245


┌ Info: Epoch 189
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6575451772260406e-5
valid_lossmean = 0.00045875010160369937
FE = 0.043958845897972505


┌ Info: Epoch 190
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.614426595021941e-5
valid_lossmean = 0.00048825886209475505
FE = 0.04535062351731463


┌ Info: Epoch 191
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6855613419919058e-5
valid_lossmean = 0.0003597206551340644
FE = 0.03892609901335047


┌ Info: Epoch 192
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.657364348645909e-5
valid_lossmean = 0.00039862713660434045
FE = 0.04097713826668099


┌ Info: Epoch 193
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.628841547248369e-5
valid_lossmean = 0.0004170133070612154
FE = 0.04191149494484119


┌ Info: Epoch 194
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6231701285094758e-5
valid_lossmean = 0.0004704045699977923
FE = 0.044513727339552396


┌ Info: Epoch 195
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.5653097759902767e-5
valid_lossmean = 0.00043217986747157925
FE = 0.04266683822097489


┌ Info: Epoch 196
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.617559816522891e-5
valid_lossmean = 0.0004181050472110437
FE = 0.04196632119684359


┌ Info: Epoch 197
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.5599586200969416e-5
valid_lossmean = 0.0004979036638892533
FE = 0.04579634898516034


┌ Info: Epoch 198
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6818808751207033e-5
valid_lossmean = 0.0005073549984776466
FE = 0.04622896463115461


┌ Info: Epoch 199
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.6195499294460672e-5
valid_lossmean = 0.0005949161715017871
FE = 0.05005945099491915


┌ Info: Epoch 200
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 1.5092391685186291e-5
valid_lossmean = 0.0004181320856671735
FE = 0.041967678135964165
107.586316 seconds (372.97 M allocations: 42.558 GiB, 10.08% gc time, 1.05% compilation time)


In [5]:
dFE_1d(comm, commModel, commLeader, m, valid=test)

FE = 0.04814524309206283


0.04814524309206283

## PEDS result

The fractional error on the test set is 0.0481.

# Baseline 

In [6]:
namemodel = "local_MSE_parallel_baselineparameterized_$endname"

##define same AL parameters for all workers
al1 = ALstruct(Ninit=ninit+8*kval, T=0, η=1e-3, J=Jval, ne=number_epoch, batchsize=batchsize);
if MPI.Comm_rank(comm) == 0
    @show kval
    @show 256+8*kval
end
##ititialize DataRunner and DataSet
dr = DataRunner(Xt, yt, [1]);
# dr = DataRunner(generateX(Xt), yt, [1]);
ds = DataSet()
validation_fes = []
##initialize baseline
mgen =  Chain(Dense(nin, nnodes, relu),
Dropout(0.5),
#BatchNorm(256, relu),
Dense(nnodes, nnodes, relu),
Dropout(0.5),
#BatchNorm(256, relu),
Dense(nnodes, coarseresolution^2, relu),
#BatchNorm(coarseresolution^2, hardtanh),
Dense(coarseresolution^2, 1, hardtanh),
x-> @. x[:]*0.9/2 + 0.45 + .1)

cw=[cwval]

m(p) = begin
    mgen(p)
end
mloglik(p) = 1
ps = Flux.params(mgen)


loader = initloader(al1, dr, ds);
opt = ADAM(al1.η)

if MPI.Comm_rank(comm)==0
    @time Flux.@epochs al1.ne train_distributed!(comm, commModel, commLeader, mloglik, m, dHuber, ps, loader, opt, validation_fes, logging=true)
else
    Flux.@epochs al1.ne train_distributed!(comm, commModel, commLeader, mloglik, m, dHuber, ps, loader, opt, validation_fes, logging=true)
end


# ##save models and validation FEs
# if isleader
#     BSON.@save "$(namemodel)_K$(kval)_mgen$(model_color).bson" mgen
# end

# if MPI.Comm_rank(comm) == 0
#     writedlm("$(namemodel)_K$(kval)_trainingMSE.csv", validation_fes, ',')
# end

# # evaluate and show the error on the test set
# valid_lossmean = dFE_1d(comm, commModel, commLeader, m)
# if MPI.Comm_rank(comm) == 0
#     @show valid_lossmean
#     writedlm("$(namemodel)_K$(kval)_endvalidFE.csv", [valid_lossmean], ',')
# end



kval = 128
256 + 8kval = 1280


┌ Info: Epoch 1
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 9.087440830678339e-5
valid_lossmean = 0.007655771921302229
FE = 0.17957781724323424
train_lossmean = 7.882662053470223e-5
valid_lossmean = 0.009459196166458565
FE = 0.19961138645332263


┌ Info: Epoch 2
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 3
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 7.351658493128562e-5
valid_lossmean = 0.00813197924014964
FE = 0.18507865118120526


┌ Info: Epoch 4
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 7.198758713021466e-5
valid_lossmean = 0.00852736828276333
FE = 0.18952465077520197
train_lossmean = 7.066432631718027e-5
valid_lossmean = 0.008297821996089872
FE = 0.18695636360681112


┌ Info: Epoch 5
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 6
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.812702824206355e-5
valid_lossmean = 0.006705730711818246
FE = 0.1680665362772487
train_lossmean = 6.749143956798479e-5
valid_lossmean = 

┌ Info: Epoch 7
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.00802158974523053
FE = 0.18381816167787127
train_lossmean = 6.47152573218137e-5
valid_lossmean = 0.006731023245485139
FE = 0.16838319287700385


┌ Info: Epoch 8
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 9
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.758362986493981e-5
valid_lossmean = 0.007296102146245177
FE = 0.17530877211983056
train_lossmean = 6.494892527978281e-5
valid_lossmean = 0.006783798408053492


┌ Info: Epoch 10
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.16904201541147604
train_lossmean = 6.172235612539465e-5
valid_lossmean = 0.00676329092123302
FE = 0.16878631421183304


┌ Info: Epoch 11
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 12
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.10655540220425e-5
valid_lossmean = 0.0066210600257699224
FE = 0.16700210977729513
train_lossmean = 5.923852973738217e-5
valid_lossmean = 0.00614359797539296
FE = 0.1608679729547632


┌ Info: Epoch 13
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 14
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 6.032710767068914e-5
valid_lossmean = 0.006667145258741248
FE = 0.16758230276184222
train_lossmean = 5.654760688235493e-5
valid_lossmean = 0.005796800128964548


┌ Info: Epoch 15
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.15626163291521128
train_lossmean = 5.798026556112128e-5
valid_lossmean = 0.006651817886944011
FE = 0.1673895610318115


┌ Info: Epoch 16
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 17
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.6960598397734e-5
valid_lossmean = 0.006776122433885633
FE = 0.16894635149826012


┌ Info: Epoch 18
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.6021832941252725e-5
valid_lossmean = 0.007239792353223907
FE = 0.1746309635400786
train_lossmean = 5.6333973788794003e-5
valid_lossmean = 0.006168172835165101
FE = 0.16118939391989395


┌ Info: Epoch 19
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 20
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.454559111161056e-5
valid_lossmean = 0.006917544169999853
FE = 0.17070025331142769
train_lossmean = 5.470851258984598e-5
valid_lossmean = 0.006452897590769332
FE = 0.1648677008430264


┌ Info: Epoch 21
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 22
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.4490994865634205e-5
valid_lossmean = 0.008085705819998817
FE = 0.18455132321011455
train_lossmean = 5.205707625380725e-5
valid_lossmean = 0.006177052617495387


┌ Info: Epoch 23
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.1613053773724892
train_lossmean = 5.194400441547726e-5
valid_lossmean = 0.006272707413077298
FE = 0.16254952713150372


┌ Info: Epoch 24
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 25
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 5.05938338900017e-5
valid_lossmean = 0.006613847224161357
FE = 0.16691112122622134
train_lossmean = 5.04287806764615e-5
valid_lossmean = 

┌ Info: Epoch 26
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.007620808914278709
FE = 0.17916729265148368
train_lossmean = 4.887116157678309e-5
valid_lossmean = 0.007457593803555511
FE = 0.1772382925385393


┌ Info: Epoch 27
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 28
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.841215912187866e-5
valid_lossmean = 0.0063769742571986475
FE = 0.16389493299544527
train_lossmean = 5.068346078047971e-5
valid_lossmean = 0.006895149014052648
FE = 0.17042371312681087


┌ Info: Epoch 29
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 30
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.958450712024888e-5
valid_lossmean = 0.0066145927832997565
FE = 0.16692052865586962
train_lossmean = 4.6514054376336356e-5
valid_lossmean = 0.007144176374194866


┌ Info: Epoch 31
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.17347395461974793
train_lossmean = 4.736115388852348e-5
valid_lossmean = 0.007465103496491383
FE = 0.17732750832859107


┌ Info: Epoch 32
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 33
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.607409215080247e-5
valid_lossmean = 0.0063606038691842864
FE = 0.1636844297335778
train_lossmean = 4.6018396528356664e-5
valid_lossmean = 0.0069027510535067445
FE = 0.17051763501942283


┌ Info: Epoch 34
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 35
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.8762732351654904e-5
valid_lossmean = 0.0075326304051670035
FE = 0.17812772638082858
train_lossmean = 4.415391859240634e-5
valid_lossmean = 0.007390389025698714


┌ Info: Epoch 36
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.17643788580964018
train_lossmean = 4.465675445119471e-5
valid_lossmean = 0.008420678121996598
FE = 0.18833530017525002


┌ Info: Epoch 37
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 38
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.430353731248956e-5
valid_lossmean = 0.007688420052895361
FE = 0.1799603157672478
train_lossmean = 4.4732951235703526e-5
valid_lossmean = 0.006129055873001502


┌ Info: Epoch 39
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.16067747021358267
train_lossmean = 4.2198007472084656e-5
valid_lossmean = 0.0066521569733159385
FE = 0.16739382744413528


┌ Info: Epoch 40
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 41
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.4024138730551824e-5
valid_lossmean = 0.006730841675136084
FE = 0.16838092178090341
train_lossmean = 4.320994564444546e-5
valid_lossmean = 0.005935696805546565
FE = 0.15812263746911379


┌ Info: Epoch 42
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 43
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.251536243884403e-5
valid_lossmean = 0.005440543839399468
FE = 0.15138378244199882
train_lossmean = 4.272677948898795e-5
valid_lossmean = 0.00697254416830946


┌ Info: Epoch 44
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.17137751143145152
train_lossmean = 4.099288172795012e-5
valid_lossmean = 0.006143114028982134
FE = 0.16086163684588703


┌ Info: Epoch 45
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 46
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 4.234193310997121e-5
valid_lossmean = 0.007155314109514102
FE = 0.17360912447280763
train_lossmean = 4.03740019990191e-5
valid_lossmean = 

┌ Info: Epoch 47
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.006382366505293288
FE = 0.16396421157180155
train_lossmean = 3.897447480322192e-5
valid_lossmean = 0.006796862309009127
FE = 0.16920470347019184


┌ Info: Epoch 48
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 49
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.9062295263741726e-5
valid_lossmean = 0.007184540263146579
FE = 0.17396331970858486
train_lossmean = 4.228491666995319e-5
valid_lossmean = 0.005631075111218554
FE = 0.15401174986532662


┌ Info: Epoch 50
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 51
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.9644060330980895e-5
valid_lossmean = 0.006425426754041045
FE = 0.1645163947739181
train_lossmean = 3.8915782889628874e-5
valid_lossmean = 0.006774360528531872


┌ Info: Epoch 52
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.1689243856294128
train_lossmean = 3.966443785261765e-5
valid_lossmean = 0.005522716519290937
FE = 0.15252273027857238


┌ Info: Epoch 53
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 54
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.758990688826725e-5
valid_lossmean = 0.005388611167566746
FE = 0.1506595335699824
train_lossmean = 3.6892346771667925e-5
valid_lossmean = 

┌ Info: Epoch 55
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.005755362502034535
FE = 0.15570212385342153
train_lossmean = 3.577327150328002e-5
valid_lossmean = 0.005891873913522808
FE = 0.15753785115441726


┌ Info: Epoch 56
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 57
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.768795646077007e-5
valid_lossmean = 0.006144758015194507
FE = 0.1608831598569563
train_lossmean = 3.7899216171269176e-5
valid_lossmean = 0.005782046776641906
FE = 0.15606265661860017


┌ Info: Epoch 58
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 59
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.681963874356753e-5
valid_lossmean = 0.005513634971151041
FE = 0.15239727458906369
train_lossmean = 3.624296955438141e-5
valid_lossmean = 0.005601545572855234


┌ Info: Epoch 60
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.15360739778346053
train_lossmean = 3.57303421628158e-5
valid_lossmean = 0.006094944874440561
FE = 0.1602297245697737


┌ Info: Epoch 61
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 62
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.485428793896544e-5
valid_lossmean = 0.005417483409003809
FE = 0.1510626121347306
train_lossmean = 3.4315105680991565e-5
valid_lossmean = 

┌ Info: Epoch 63
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.005466388628048871
FE = 0.15174292359704286
train_lossmean = 3.506904491103749e-5
valid_lossmean = 0.004201864288043962
FE = 0.13303907169933177


┌ Info: Epoch 64
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 65
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.514388238903252e-5
valid_lossmean = 0.004486618024240302
FE = 0.13747310552160655
train_lossmean = 3.426457675865112e-5
valid_lossmean = 0.004660369748839257
FE = 0.14010975782476567


┌ Info: Epoch 66
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 67
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.327841681585514e-5
valid_lossmean = 0.005503952680736732
FE = 0.15226340618466314
train_lossmean = 3.420472488663018e-5
valid_lossmean = 0.005988675687179486


┌ Info: Epoch 68
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.1588267292718273
train_lossmean = 3.3180322653696395e-5
valid_lossmean = 0.005432748372388367
FE = 0.15127528866377424


┌ Info: Epoch 69
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 70
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.271353270075576e-5
valid_lossmean = 0.0049223008006497706
FE = 0.1439932950876264
train_lossmean = 3.1985433637214284e-5
valid_lossmean = 0.0047342422092724705


┌ Info: Epoch 71
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.14121584586101785
train_lossmean = 3.317309178775916e-5
valid_lossmean = 0.005291580977429394
FE = 0.14929694410262143


┌ Info: Epoch 72
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 73
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.16308472532089e-5
valid_lossmean = 0.005275033452034823
FE = 0.14906332494349034
train_lossmean = 3.2771388266056095e-5
valid_lossmean = 0.005463767134436018
FE = 0.15170653386741348


┌ Info: Epoch 74
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 75
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.195742941468816e-5
valid_lossmean = 0.0047187086732636396
FE = 0.14098398365063133
train_lossmean = 3.155393403784606e-5
valid_lossmean = 0.005889612455827643


┌ Info: Epoch 76
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.1575076146470195
train_lossmean = 3.1899591190090786e-5
valid_lossmean = 0.005942226074701335
FE = 0.1582095810523438


┌ Info: Epoch 77
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 78
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.086983292882121e-5
valid_lossmean = 0.005293772923876054
FE = 0.1493278627472591
train_lossmean = 3.1495359233515435e-5
valid_lossmean = 

┌ Info: Epoch 79
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0045960327229607745
FE = 0.13913927984434793
train_lossmean = 3.03742162525568e-5
valid_lossmean = 0.004931441073983595
FE = 0.14412692443165626


┌ Info: Epoch 80
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 81
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.1239783540927667e-5
valid_lossmean = 0.005409263600387905
FE = 0.1509479669178726
train_lossmean = 3.1473389286404404e-5
valid_lossmean = 0.005022439527551551
FE = 0.14545061213071317


┌ Info: Epoch 82
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 83
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.9534354654966086e-5
valid_lossmean = 0.004060242538132699
FE = 0.13077784709240314
train_lossmean = 2.9179854650763242e-5
valid_lossmean = 

┌ Info: Epoch 84
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.004564616019947378
FE = 0.13866291321880175
train_lossmean = 3.0278753413412628e-5
valid_lossmean = 0.003980713025527929
FE = 0.1294907153585132


┌ Info: Epoch 85
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 86
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 3.0133378292253047e-5
valid_lossmean = 0.003890307159486318
FE = 0.12801184039273933
train_lossmean = 3.018723845487449e-5
valid_lossmean = 0.004574379808844641


┌ Info: Epoch 87
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.13881113512722296
train_lossmean = 3.0157307541764176e-5
valid_lossmean = 0.004732943893702622
FE = 0.14119648106144084


┌ Info: Epoch 88
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 89
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.879536291340158e-5
valid_lossmean = 0.004873697687176215
FE = 0.14328063199250848
train_lossmean = 2.9593520504327474e-5
valid_lossmean = 0.004668948713006279
FE = 0.14023865789024098


┌ Info: Epoch 90
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 91
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.9474244627444653e-5
valid_lossmean = 0.0040887836257916576
FE = 0.13123668736231697
train_lossmean = 2.937192370405399e-5
valid_lossmean = 

┌ Info: Epoch 92
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0035130706775456234
FE = 0.12164706602462587
train_lossmean = 2.8785971514500273e-5
valid_lossmean = 0.003874670606241173
FE = 0.1277543183923415


┌ Info: Epoch 93
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 94
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.8739639197055014e-5
valid_lossmean = 0.004804954938825615
FE = 0.14226656796793832
train_lossmean = 2.801254155452165e-5
valid_lossmean = 0.004555866761315438


┌ Info: Epoch 95
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.13852995793908626
train_lossmean = 2.8009032977133345e-5
valid_lossmean = 0.004293199942917533
FE = 0.13447722931877132


┌ Info: Epoch 96
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 97
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.8968714180060805e-5
valid_lossmean = 0.00412592225575784
FE = 0.1318313548245006
train_lossmean = 2.8431247267996832e-5
valid_lossmean = 0.003944020639305357
FE = 0.1288925407422158


┌ Info: Epoch 98
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 99
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.7612503516353526e-5
valid_lossmean = 0.0037601113029854688
FE = 0.12585154326130601
train_lossmean = 2.7263889695491277e-5
valid_lossmean = 

┌ Info: Epoch 100
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.003647552179334036
FE = 0.12395354498692943
train_lossmean = 2.741603296156261e-5
valid_lossmean = 0.004515505873958108
FE = 0.1379149673440832


┌ Info: Epoch 101
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 102
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.7893481465273227e-5
valid_lossmean = 0.005230284736915704
FE = 0.14842971768086585
train_lossmean = 2.891579418861072e-5
valid_lossmean = 0.003902056671712812
FE = 0.12820500544117955


┌ Info: Epoch 103
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 104
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6565767478732026e-5
valid_lossmean = 0.0036963320419510872
FE = 0.12477962716840238


┌ Info: Epoch 105
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.670621542848142e-5
valid_lossmean = 0.004217370471088549
FE = 0.13328432337203158
train_lossmean = 2.56644241390505e-5
valid_lossmean = 0.004251014892194765
FE = 0.1338149106198547


┌ Info: Epoch 106
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 107
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.733643198588258e-5
valid_lossmean = 0.0042879352511886295
FE = 0.13439475024181294


┌ Info: Epoch 108
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.553612205779565e-5
valid_lossmean = 0.0041926196883825375
FE = 0.13289264024705183
train_lossmean = 2.6673765992997147e-5
valid_lossmean = 0.004102186347268321
FE = 0.13145160332241862


┌ Info: Epoch 109
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 110
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6484573141932713e-5
valid_lossmean = 0.00357688064696721
FE = 0.12274686802294027
train_lossmean = 2.5925147247567665e-5
valid_lossmean = 0.003104424100316233
FE = 0.11435330429995041


┌ Info: Epoch 111
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 112
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6749223955624042e-5
valid_lossmean = 0.0035612791792541463
FE = 0.12247887975645168
train_lossmean = 2.5947147018103198e-5
valid_lossmean = 

┌ Info: Epoch 113
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0033362617739967717
FE = 0.11854636859749075
train_lossmean = 2.6607306342364495e-5
valid_lossmean = 0.0040381179948940655
FE = 0.13042105159735018


┌ Info: Epoch 114
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 115
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.7289493005288756e-5
valid_lossmean = 0.0037451410878414255
FE = 0.1256007656251522
train_lossmean = 2.439709529024369e-5
valid_lossmean = 

┌ Info: Epoch 116
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.003629751990310825
FE = 0.1236507261601457
train_lossmean = 2.5196546466368447e-5
valid_lossmean = 0.0036461007994876148
FE = 0.1239288816554742


┌ Info: Epoch 117
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 118
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.6130686364119717e-5
valid_lossmean = 0.0032970901010325702
FE = 0.11784837619951173
train_lossmean = 2.4968310394584024e-5
valid_lossmean = 0.003868546902033978
FE = 0.12765332412878513


┌ Info: Epoch 119
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 120
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.506665867631686e-5
valid_lossmean = 0.003991393135727839
FE = 0.1296643084717549
train_lossmean = 2.503497309077519e-5
valid_lossmean = 

┌ Info: Epoch 121
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.00383769341143571
FE = 0.1271432572795102
train_lossmean = 2.5516142946949197e-5
valid_lossmean = 0.003988958912025631
FE = 0.1296247633730354


┌ Info: Epoch 122
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 123
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4011021398645423e-5
valid_lossmean = 0.0034093849341326897
FE = 0.11983845849993863


┌ Info: Epoch 124
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4737276690212802e-5
valid_lossmean = 0.00426456715724976
FE = 0.13402804232175788
train_lossmean = 2.51146307848246e-5
valid_lossmean = 0.003933805445151592


┌ Info: Epoch 125
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12872551372727958
train_lossmean = 2.4663514088645396e-5
valid_lossmean = 0.003763265169679066


┌ Info: Epoch 126
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12590431241661215
train_lossmean = 2.3905541697795363e-5
valid_lossmean = 0.003590572378691067
FE = 0.12298157140549809


┌ Info: Epoch 127
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 128
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4516055217735573e-5
valid_lossmean = 0.0033435213816586655
FE = 0.11867527522001248


┌ Info: Epoch 129
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4000343198748765e-5
valid_lossmean = 0.003797839647550977
FE = 0.1264813542957189
train_lossmean = 2.4792079367321877e-5
valid_lossmean = 0.0038280633213677723


┌ Info: Epoch 130
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12698363404184723
train_lossmean = 2.3038733810767683e-5
valid_lossmean = 0.0035989783513441444
FE = 0.12312544475950193


┌ Info: Epoch 131
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 132
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3855812812848014e-5
valid_lossmean = 0.0038993388345230076
FE = 0.12816034937188578
train_lossmean = 2.5707877233463895e-5
valid_lossmean = 0.003857811983682357
FE = 0.12747608703947547


┌ Info: Epoch 133
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 134
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.4066769016591068e-5
valid_lossmean = 0.0034463479163823444
FE = 0.12048632397388928
train_lossmean = 2.3890216873573543e-5
valid_lossmean = 0.003139988977835194
FE = 0.11500646562796316


┌ Info: Epoch 135
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 136
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.400986622656027e-5
valid_lossmean = 0.003985737518744431
FE = 0.12957241178323728
train_lossmean = 2.2314254864961052e-5
valid_lossmean = 0.0031595123352798397


┌ Info: Epoch 137
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.11536344665855522
train_lossmean = 2.434951829690428e-5
valid_lossmean = 0.003947261896330386
FE = 0.12894549280476253


┌ Info: Epoch 138
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 139
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.378644293386315e-5
valid_lossmean = 0.0034764310363863144
FE = 0.1210110430657815
train_lossmean = 2.4001859509544422e-5
valid_lossmean = 

┌ Info: Epoch 140
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0040996087273285405
FE = 0.13141029784670083
train_lossmean = 2.4774027651624995e-5
valid_lossmean = 0.004675563339000691
FE = 0.14033796268165305


┌ Info: Epoch 141
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 142
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3891838523620472e-5
valid_lossmean = 0.0042577582852005675
FE = 0.13392100398606693
train_lossmean = 2.370257956110385e-5
valid_lossmean = 0.004153958799016055
FE = 0.13227850791570311


┌ Info: Epoch 143
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 144
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.425697531224799e-5
valid_lossmean = 0.003081458339478589
FE = 0.11392954040437284
train_lossmean = 2.252951829285229e-5
valid_lossmean = 0.003849096234528366


┌ Info: Epoch 145
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12733200564746544
train_lossmean = 2.281962091127626e-5
valid_lossmean = 0.003565607725958398
FE = 0.12255329045671462


┌ Info: Epoch 146
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 147
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3253657183481807e-5
valid_lossmean = 0.0041169203928936365
FE = 0.13168746265856232
train_lossmean = 2.374291886392105e-5
valid_lossmean = 

┌ Info: Epoch 148
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.004633982377103107
FE = 0.13971253864606434
train_lossmean = 2.5026157158821378e-5
valid_lossmean = 0.004140605082641801
FE = 0.13206571914398316


┌ Info: Epoch 149
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 150
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.394444965571933e-5
valid_lossmean = 0.0033138062847511227
FE = 0.11814674300522343
train_lossmean = 2.3715203071480048e-5
valid_lossmean = 0.003083551602348838
FE = 0.11396823052493629


┌ Info: Epoch 151
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 152
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3721150790875377e-5
valid_lossmean = 0.0037871042683312534
FE = 0.12630246493613928
train_lossmean = 2.3685929715444955e-5
valid_lossmean = 

┌ Info: Epoch 153
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0035210289394554486
FE = 0.12178477341155919
train_lossmean = 2.3295868552955343e-5
valid_lossmean = 0.0036747269332422635
FE = 0.12441442346191497


┌ Info: Epoch 154
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 155
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.328626799305761e-5
valid_lossmean = 0.0032986853904349617
FE = 0.11787688308350104
train_lossmean = 2.3182653881698945e-5
valid_lossmean = 

┌ Info: Epoch 156
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.004074167849330612
FE = 0.13100191787255816
train_lossmean = 2.3553074808754555e-5
valid_lossmean = 0.0035540717129642968
FE = 0.12235487803571514


┌ Info: Epoch 157
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 158
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3169054057000332e-5
valid_lossmean = 0.0027505556301536675
FE = 0.10763869199248734
train_lossmean = 2.2621960213583872e-5
valid_lossmean = 0.003497136587423767
FE = 0.12137087769868594


┌ Info: Epoch 159
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 160
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2738042870647683e-5
valid_lossmean = 0.0035529595862005413
FE = 0.12233573312360481


┌ Info: Epoch 161
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2953606888237932e-5
valid_lossmean = 0.003899631098769114
FE = 0.128165152235607
train_lossmean = 2.2573775075519315e-5
valid_lossmean = 0.0038533730086462797
FE = 0.12740272602017347


┌ Info: Epoch 162
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 163
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2490757117369866e-5
valid_lossmean = 0.004170413039205356
FE = 0.13254023309521737


┌ Info: Epoch 164
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.177907731741926e-5
valid_lossmean = 0.004096248724909369
FE = 0.13135643545940157
train_lossmean = 2.24625059652075e-5
valid_lossmean = 0.004119395527694686
FE = 0.1317270426376553


┌ Info: Epoch 165
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 166
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.309498638447647e-5
valid_lossmean = 0.0041243721224211655
FE = 0.13180658758904076
train_lossmean = 2.1149434231135106e-5
valid_lossmean = 0.0036804054914052736
FE = 0.12451051520244796


┌ Info: Epoch 167
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 168
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2279935102501275e-5
valid_lossmean = 0.0034942935815265234
FE = 0.12132153328935162
train_lossmean = 2.1694216745702667e-5
valid_lossmean = 

┌ Info: Epoch 169
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.003343942290329809
FE = 0.11868274487172359
train_lossmean = 2.2165936834063737e-5
valid_lossmean = 0.003512897759801609
FE = 0.12164407217721977


┌ Info: Epoch 170
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 171
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2565117016142915e-5
valid_lossmean = 0.003786917697883609
FE = 0.12629935377310003
train_lossmean = 2.356204229376645e-5
valid_lossmean = 0.0033086281407788925
FE = 0.11805439904012924


┌ Info: Epoch 172
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 173
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.24669596700735e-5
valid_lossmean = 0.003037411120910158
FE = 0.11311233937342113


┌ Info: Epoch 174
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.3612273282577564e-5
valid_lossmean = 0.00337963088269335
FE = 0.11931439134807229
train_lossmean = 2.2242961268228474e-5
valid_lossmean = 0.0036356770818615457
FE = 0.12375160676207526


┌ Info: Epoch 175
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 176
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.147407727919219e-5
valid_lossmean = 0.0034009379159900413
FE = 0.11968991185151012
train_lossmean = 2.1233480733347083e-5
valid_lossmean = 0.003627601905316036


┌ Info: Epoch 177
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12361409845751047
train_lossmean = 2.136497218516662e-5
valid_lossmean = 0.0032470378098682015
FE = 0.11695044213883875


┌ Info: Epoch 178
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 179
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.2082970466672416e-5
valid_lossmean = 0.00373205697937358
FE = 0.12538117284516054
train_lossmean = 2.1840499157165868e-5
valid_lossmean = 0.0037431437714949864
FE = 0.1255672691676896


┌ Info: Epoch 180
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 181
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0971830874259432e-5
valid_lossmean = 0.0033907600044757657
FE = 0.11951068098292654


┌ Info: Epoch 182
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.145512070382059e-5
valid_lossmean = 0.0035251681356299096
FE = 0.12185633530667817
train_lossmean = 2.1618219832570603e-5
valid_lossmean = 0.004061635939881964
FE = 0.1308002854624883


┌ Info: Epoch 183
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 184
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.155896853616143e-5
valid_lossmean = 0.003891150873692828
FE = 0.12802572098733034
train_lossmean = 2.1923229994209245e-5
valid_lossmean = 0.004249485958068608


┌ Info: Epoch 185
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.1337908442982599
train_lossmean = 2.1123149539586535e-5
valid_lossmean = 0.003316713692204008
FE = 0.11819856036886175


┌ Info: Epoch 186
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 187
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0957871831089384e-5
valid_lossmean = 0.003688859859344155
FE = 0.12465344155673337
train_lossmean = 2.1352712131395734e-5
valid_lossmean = 0.0034300172853464666


┌ Info: Epoch 188
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


FE = 0.12020052082901642
train_lossmean = 2.170340341969431e-5
valid_lossmean = 0.003492162854115453
FE = 0.1212845383246179


┌ Info: Epoch 189
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 190
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.190769694479587e-5
valid_lossmean = 0.004009064909009788
FE = 0.12995103387177653
train_lossmean = 2.241570466241422e-5
valid_lossmean = 0.003854404100204366
FE = 0.1274197701901149


┌ Info: Epoch 191
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 192
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.181486885723247e-5
valid_lossmean = 0.0033670209002810707
FE = 0.11909159210034542
train_lossmean = 2.1623486047405177e-5
valid_lossmean = 

┌ Info: Epoch 193
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


0.0033424673503322437
FE = 0.1186565677999971
train_lossmean = 2.1544432213407326e-5
valid_lossmean = 0.004060036151290502
FE = 0.1307745232552629


┌ Info: Epoch 194
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 195
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.0773772357351885e-5
valid_lossmean = 0.003454288553460624
FE = 0.12062504875300405
train_lossmean = 2.1402792660951263e-5
valid_lossmean = 0.0029102168502632086
FE = 0.11071867265551691


┌ Info: Epoch 196
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 197
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1766606287856584e-5
valid_lossmean = 0.003759685576469673
FE = 0.12584441849140598


┌ Info: Epoch 198
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1700764215927183e-5
valid_lossmean = 0.004076829577912635
FE = 0.13104470386333522
train_lossmean = 2.20650755241679e-5
valid_lossmean = 0.003203319350285856
FE = 0.11616045747354867


┌ Info: Epoch 199
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154
┌ Info: Epoch 200
└ @ Main /Users/raphaelpestourie/.julia/packages/Flux/EXOFx/src/optimise/train.jl:154


train_lossmean = 2.1464521044725547e-5
valid_lossmean = 0.0036031543992685916
FE = 0.12319685789939004
 18.248528 seconds (7.97 M allocations: 9.089 GiB, 9.29% gc time, 9.73% compilation time)


In [7]:
dFE_1d(comm, commModel, commLeader, m, valid=test)

FE = 0.13068358691983506


0.13068358691983506

## Baseline result 

The fractional error on the test set is 0.13.

# Overall result

With about 1000 data points, a single model PEDS model achieves an error < 5% and leads to a 2.7x improvement compared to the NN-only baseline. 